### PR TITLE
Add ability to reuse same component multiple times in the same view

### PR DESCRIPTION
**Pull request by mskorkowski**
Solution for #25

### CODE REVIEW

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Thanks a lot! Looks pretty good overall, just enabling the tracker features again would be nice, most notable un-commenting all places where `self.reset()` was used.

File: relm4-components/src/alert.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ pub struct AlertSettings {
 pub struct AlertModel<Conf: AlertConfig> {
     settings: AlertSettings,
     is_active: bool,
-    _conf_provider: PhantomData<Conf>,
+    _conf_provider: PhantomData<*const Conf>,
```
[the remark this answers is not in the document]
Comment (mskorkowski) [reply relm4-components/src/alert.rs:34]: As PahntomData doc says `PhantomData<T>` indicates ownership of T, and `PhantomData<&'a T>` or `PhantomData<*const T>` does not.

https://docs.rs/rustc-std-workspace-std/1.0.1/std/marker/struct.PhantomData.html#ownership-and-the-drop-check

File: src/lib.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #![doc(
     html_favicon_url = "https://raw.githubusercontent.com/AaronErhardt/relm4/main/assets/Relm_logo.svg"
 )]
+#![feature(doc_cfg)]
```
[the remark this answers is not in the document]
Comment (mskorkowski) [reply src/lib.rs:15]: Of course. It was complaining while I was compiling doc comments

[the remark this answers is not in the document]
Comment (AaronErhardt) [reply src/lib.rs]: Yeah the docs are a bit weird right now. They only work in nightly and are deployed by a custom script. Hopefully this feature will be stabilised soon.

**Comment by mskorkowski** [comment]
All should be fixed

**Review by AaronErhardt — APPROVED** [review]
Thanks a lot for your contribution!!

**Pull request merged**